aea/helpers/async_utils.py

```diff
@@ -127,7 +127,7 @@ async def wait(self, state_or_states: Union[Any, Sequence[Any]]) -> Tuple[Any, A
             return (None, self._state)
 
         watcher: Future = Future()
-        watcher._states = states  # type: ignore
+        watcher._states = states  # type: ignore  # pylint: disable=protected-access
```

Comment:
I don't think there are counterarguments in making the `_states` attribute to be public. It is a custom attribute, so it won't be used by the `asyncio` library (I would be very surprised if it is).

Moreover, an ideal approach would be to extend the `Future` class to include such custom attributes.

Reply:
This, of course, will affect the above code, which must be changed accordingly. But that would save some Pylint `disable `directives.

Reply:
Good points; although I'd rather not address them now as I don't want to open a can of worms.